### PR TITLE
Introduce Event API Definition entity

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- Add EventAPI entity
- Add EventAPI converter from model to entity
- Simplify EventApi fixtures

**Related issue(s)**

see #223

### CODE REVIEW

File: components/director/internal/domain/eventapi/converter.go [hunk 1]
```diff
@@ -109,3 +115,99 @@ func (c *converter) eventAPISpecInputFromGraphQL(in *graphql.EventAPISpecInput)
 		FetchRequest:  c.fr.InputFromGraphQL(in.FetchRequest),
 	}
 }
+
+func (c *converter) FromEntity(eventEntity EventAPIDefinition) (model.EventAPIDefinition, error) {
+	ver, err := c.convertVersionFromEntity(eventEntity.Version)
+	if err != nil {
+		return model.EventAPIDefinition{}, errors.Wrap(err, "while converting version")
+
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/converter.go]: empty line

File: components/director/internal/domain/eventapi/converter.go [hunk 2]
```diff
@@ -109,3 +115,99 @@ func (c *converter) eventAPISpecInputFromGraphQL(in *graphql.EventAPISpecInput)
...
+			return nil, errors.Wrap(err, "while converting version")
+		}
+		return &tmp, nil
+	}
+	return nil, nil
+}
+
+func (c *converter) apiSpecToEntity(spec *model.EventAPISpec) *EventAPISpec {
+	var apiSpecEnt *EventAPISpec
+	if spec != nil {
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/converter.go]: inverse the condition

File: components/director/internal/domain/eventapi/converter.go [hunk 3]
```diff
@@ -109,3 +115,99 @@ func (c *converter) eventAPISpecInputFromGraphQL(in *graphql.EventAPISpecInput)
...
+		apiSpecEnt = &tmp
+	}
+
+	return apiSpecEnt
+}
+
+func (c *converter) apiSpecFromEntity(specEnt *EventAPISpec) *model.EventAPISpec {
+	var apiSpec *model.EventAPISpec
+
+	if specEnt != nil {
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/converter.go]: inverse the condition

File: components/director/internal/domain/eventapi/converter.go [hunk 4]
```diff
@@ -109,3 +115,99 @@ func (c *converter) eventAPISpecInputFromGraphQL(in *graphql.EventAPISpecInput)
...
+		Description:        repo.NewNullableString(eventModel.Description),
+		GroupName:          repo.NewNullableString(eventModel.Group),
+		Version:            ver,
+		EventAPISpec:       c.apiSpecToEntity(eventModel.Spec),
+		SpecFetchRequestID: repo.NewNullableString(eventModel.SpecFetchRequestId),
+	}, nil
+}
+
+func (c *converter) convertVersionFromEntity(inVer *version.Version) (*model.Version, error) {
+	if inVer != nil {
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/converter.go]: inverse the condition

File: components/director/internal/domain/eventapi/converter.go [hunk 5]
```diff
@@ -109,3 +115,99 @@ func (c *converter) eventAPISpecInputFromGraphQL(in *graphql.EventAPISpecInput)
...
+		if err != nil {
+			return nil, errors.Wrap(err, "while converting version")
+		}
+		return &tmp, nil
+	}
+	return nil, nil
+}
+
+func (c *converter) convertVersionToEntity(inVer *model.Version) (*version.Version, error) {
+	if inVer != nil {
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/converter.go]: inverse the condition

File: components/director/internal/domain/eventapi/service_test.go [hunk 6]
```diff
@@ -24,11 +24,11 @@ func TestService_Get(t *testing.T) {
 	testErr := errors.New("Test error")
 
 	id := "foo"
-	appID := "bar"
-	name := "foo"
-	desc := "bar"
+	//appID := "bar"
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/service_test.go]: Remove commented code

File: components/director/internal/domain/eventapi/service_test.go [hunk 7]
```diff
@@ -93,13 +93,13 @@ func TestService_List(t *testing.T) {
 
 	id := "foo"
 	applicationID := "bar"
-	name := "foo"
-	desc := "bar"
+	//name := "foo"
+	//desc := "bar"
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/service_test.go]: Remove commented code

File: components/director/internal/domain/eventapi/entity.go [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package eventapi
+
+import (
+	"database/sql"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/version"
+)
+
+type EventAPIDefinition struct {
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/entity.go]: It should be called `Entity`

File: components/director/internal/domain/eventapi/entity.go [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+	AppID       string         `db:"app_id"`
+	Name        string         `db:"name"`
+	Description sql.NullString `db:"description"`
+	GroupName   sql.NullString `db:"group_name"`
+	*version.Version
+	*EventAPISpec
+	SpecFetchRequestID sql.NullString `db:"spec_fetch_request_id"`
+}
+
+type EventAPISpec struct {
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/entity.go]: You can rename it to `EntitySpec` or something like t hat

File: components/director/internal/domain/eventapi/converter_test.go [hunk 10]
```diff
@@ -301,3 +307,64 @@ func TestEventApiSpecDataConversionNilStaysNil(t *testing.T) {
 	require.NotNil(t, convertedGraphqlEvAPIDef)
 	assert.Nil(t, convertedGraphqlEvAPIDef.Spec.Data)
 }
+
+func TestConverter_ToEntity(t *testing.T) {
+	t.Run("success all nullable properties filled", func(t *testing.T) {
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/converter_test.go]: `success when...`?

File: components/director/internal/domain/eventapi/entity.go [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+type Entity struct {
+	ID          string         `db:"id"`
+	TenantID    string         `db:"tenant_id"`
+	AppID       string         `db:"app_id"`
+	Name        string         `db:"name"`
+	Description sql.NullString `db:"description"`
+	GroupName   sql.NullString `db:"group_name"`
+	*version.Version
+	*EntitySpec
+	SpecFetchRequestID sql.NullString `db:"spec_fetch_request_id"`
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/entity.go]: Why it isn't in the `EntitySpec`?

File: components/director/internal/domain/eventapi/fixtures_test.go [hunk 12]
```diff
@@ -133,21 +132,48 @@ func fixGQLEventAPIDefinitionInput(name, description string, group string) *grap
...
+		Version: &version.Version{
+			VersionDepracated:      repo.NewNullableBool(&boolean),
+			VersionValue:           repo.NewValidNullableString("v1.1"),
+			VersionDepracatedSince: repo.NewValidNullableString("v1.0"),
+			VersionForRemoval:      repo.NewNullableBool(&boolean),
+		},
 	}
 }
+
+func fixMinEntityEventApiDef(id, placeholder string) eventapi.Entity {
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/fixtures_test.go]: Api -> API

File: components/director/internal/domain/eventapi/fixtures_test.go [hunk 13]
```diff
@@ -133,21 +132,48 @@ func fixGQLEventAPIDefinitionInput(name, description string, group string) *grap
...
+		Name:        "name",
+		Description: strings.Ptr("description"),
+		Group:       strings.Ptr("group"),
 		Spec:        spec,
-		Version:     version,
+		Version:     v,
+	}
+}
+
+func fixFullEventApiDef(id, placeholder string) eventapi.Entity {
```
Comment (pkosiec) [inline components/director/internal/domain/eventapi/fixtures_test.go]: Api -> API

File: components/director/internal/domain/eventapi/converter_test.go [hunk 14]
```diff
@@ -301,3 +307,64 @@ func TestEventApiSpecDataConversionNilStaysNil(t *testing.T) {
...
+
+	})
+}
+
+func TestConverter_FromEntity(t *testing.T) {
+	t.Run("success when all nullable properties filled and converted", func(t *testing.T) {
+		//GIVEN
+		id := "id"
+		eventEntity := fixFullEventAPIDef(id, "placeholder")
+		versionConv := version.NewConverter()
```
Comment (tgorgol) [inline components/director/internal/domain/eventapi/converter_test.go]: Why isn't this converter mocked?

Comment (dbadura) [reply components/director/internal/domain/eventapi/converter_test.go]: Why not ; )

**Pull request merged**